lessons/advanced/strings-in-depth.md

```diff
@@ -0,0 +1,163 @@
+---
+layout: page
+title: Strings in-depth 
+category: advanced
+order: 9
+lang: en
+---
+
+String manipulation is one of our must to have skills nowadays, ignoring the programming language. That's why we are including an in-depth on Strings.
+
+## Table of Contents
+
+- [Strings in Elixir](#strings-in-elixir)
+- [Char lists](#char-lists)
+- [Graphemes and Codepoints](#graphemes)
+- [String functions](#string-functions)
+  - [Length](#length)
+  - [Replace](#replace)
+  - [Duplicate](#duplicate)
+- [Exercises](#exercises)
+  - [Anagrams](#anagrams)
+
+## Strings in Elixir
+
+Elixir strings are nothing but a sequence of bytes. Let's watch a clear example:
+
+```elixir
+iex(2)> string = <<104,101,108,108,111>>
+"hello"
+```
+
+>NOTE: Using << >> syntax we are saying to the compiler that the elements inside those symbols are bytes.
+
+On the previous example we just defined a string with a list of bytes, and the clear result was the string "hello". If we want to use a Guard in our functions to check if a parameter is a string, we just have to write:
+
+```elixir
+def parse_string(string) when is_binary(string) do
+  ...
+end
+```
+
+## Char lists
+
+Internally, Elixir strings are represented with a sequence of bytes rather than an array of characters, but it's worth to mention Elixir also has a data type that is a char list (characters list).
```

Comment:
> Internally, Elixir strings are represented with a sequence of bytes rather than an array of characters, ~~but it's worth to mention~~ Elixir also has a ~~data~~ **char list** type ~~that is a char list~~ (characters list).